pandas/io/html.py

```diff
@@ -627,17 +627,7 @@ def _build_doc(self):
                 except AttributeError:
                     pass
             else:
-                # not a url
-                scheme = parse_url(self.io).scheme
-                if scheme not in _valid_schemes:
-                    # lxml can't parse it
-                    msg = (('{invalid!r} is not a valid url scheme, valid '
```

Comment:
Rather than creating a custom error message here I re-raised. This makes the behavior consistent between lxml and bs4, allowing the `test_bad_url_protocol` and `test_invalid_url` tests to pass